clients/pkg/promtail/targets/file/filetarget.go

```diff
@@ -387,25 +408,28 @@ func (t *FileTarget) startTailing(ps []string) {
 			}
 			reader = tailer
 		}
+		t.readersMutex.Lock()
 		t.readers[p] = reader
+		t.readersMutex.Unlock()
```

Comment:
same here, we could use `getReader`?

Reply:
This is doing a `set` though.. I could add a corresponding `setReader()` function, but no such animal right now